### DISCUSSION TITLE
Filter Values Truncated

### DISCUSSION THREAD

**Discussion by dannyedgel** [bertrandmartel/tableau-scraping#43] [Q&A]
I'm scraping the public Tableau dashboard at https://public.tableau.com/views/RedfinDataCentertrend/MedianSalePrice?:showVizHome=no. I'm trying to scrape by zip code, and I have the following commands so far:

`url = ("https://public.tableau.com/views/RedfinDataCentertrend/MedianSalePrice?:showVizHome=no")`

`ts = TS()`
`ts.loads(url)`
`wb = ts.getWorkbook()`
`wsNames = wb.getWorksheetNames()`
`wbParams = wb.getParameters()`
`wb = wb.setParameter('Show Values As', 'Value')`
`ws = wb.getWorksheet(wsNames[0])`
`countyWb = ws.setFilter('Region Type', 'zip code', filterDelta=True)`
`countyWs = countyWb.getWorksheet(wsNames[0])`

I would like to supply some zip code or zip code vector, `zipCodes` to the following to obtain an `outData` DataFrame:

`outData = countyWs.setFilter('Region', zipCodes).getWorksheet(wsNames[0]).data`

However, I this only works for zip codes up to 01770. The following line works:

`outData = countyWs.setFilter('Region', ['Zip Code: 01741']).getWorksheet(wsNames[0]).data`

But if I supply _any_ zip code with a higher number, I get the following error:
`tableauScraper - ERROR - 'Zip Code: [zip]' is not in list`

I did not run into this issue when scraping counties, and the full list of zip codes is available when accessing the Tableau in a browser. Any clue as to why this is happening or idea for getting around it?

**Comment by bertrandmartel** [comment]
There is this call used to get the rest of the filter list paginated by index:

https://public.tableau.com/vizql/w/RedfinDataCentertrend/v/MedianSalePrice/getfilteritems/sessions/D891AA09E3ED4AE39651A6EB7F30FD28-0:0/sheets/med_sale_price/filters/%5Bredshift.110nhoh09zn5xk148lfdm027yacn%5D.%5Bnone%3Aregion%3Ank%5D?fromIndex=400&toIndex=599&domain=relevant

I would need to implement this API, probably I need to paginate through the indices if I don't know the max index. I will be able to work on the `getfilteritems` call next week 

As a workaround for you usecase, in v0.1.29, I've added an `indexValues` property in `setFilter` method, so you can manually set the indices:

```python
from tableauscraper import TableauScraper as TS

url = "https://public.tableau.com/views/RedfinDataCentertrend/MedianSalePrice?:showVizHome=no"

ts = TS()
ts.loads(url)
wb = ts.getWorkbook()
wsNames = wb.getWorksheetNames()
wbParams = wb.getParameters()
wb = wb.setParameter('Show Values As', 'Value')
ws = wb.getWorksheet(wsNames[0])
countyWb = ws.setFilter('Region Type', 'zip code', filterDelta=True)
countyWs = countyWb.getWorksheet(wsNames[0])
print(countyWs.getFilters())
outData = countyWs.setFilter('Region',
                             [], indexValues=[200, 201, 202, 203], filterDelta=True).getWorksheet(wsNames[0]).data
print(outData)
```
You can get the index value by checking the value from the network tabs of Chrome dev console:

![indices](https://user-images.githubusercontent.com/5183022/141210281-a3589ff6-0aa8-47c6-b010-f3dbd2af4565.png)

Note that the indices depend on the former filters (eg in this case `ws.setFilter('Region Type', 'zip code', filterDelta=True)`)

**Reply by dannyedgel** [reply]
Thanks! a getFilterItems() call would be amazing and would solve so many of my problems with this project.

With the new tableauscraper update, I now get an error with `from tableauscraper import TableauScraper as TS`:

`ModuleNotFoundError: No module named 'pandas._libs.interval'`

I updated pandas and still get the error. I looked in pandas, and Interval is there. Not sure what the issue is here.

**Reply by dannyedgel** [reply]
Hello again,

What is the status of the getFilterItems() call? If you're still planning to add it, I'm going to put this project on hold until then. Otherwise, I'm moving forward with a more time- and code-intensive workaround.

Thank you!
